pyEpiabm/pyEpiabm/tests/test_sweep/test_update_place_sweep.py

```diff
@@ -1,54 +1,95 @@
         """Tests that the update place sweep correctly binds
         the given population.
         """
-        test_sweep = pe.sweep.UpdatePlaceSweep()
-        test_sweep.bind_population(self.pop)
+        test_pop = self.pop
+        test_sweep = UpdatePlaceSweep()
+        test_sweep.bind_population(test_pop)
         self.assertEqual(test_sweep._population.cells[0].
-                         places[0].place_type, pe.property.PlaceType.Hotel)
+                         places[0].place_type, PlaceType.Workplace)
 
-    def test__call__(self):
+    @mock.patch('logging.warning')
+    def test_update_place(self, log_mock):
+        """Test explicitly the update place function.
+        """
+        test_pop = self.pop
+        place = test_pop.cells[0].places[0]
+        person = test_pop.cells[0].persons[0]
+        test_sweep = UpdatePlaceSweep()
+        test_sweep.bind_population(test_pop)
+        test_sweep.update_place_group(place)
+        self.assertTrue(place.persons)
+        self.assertDictEqual(place.person_groups, {0: [person]})
+        self.place.empty_place()
+        test_sweep.update_place_group(place, person_list=[person],
+                                      group_index=1)
+        self.assertDictEqual(place.person_groups,
+                             {0: [], 1: [person]})
+        self.place.empty_place([1])
+        test_sweep.update_place_group(place, person_list=[person],
+                                      person_weights=[1], group_index=1)
+        self.assertDictEqual(place.person_groups,
+                             {0: [], 1: [person]})
+        self.place.empty_place()
+        test_sweep.update_place_group(place, person_list=[])
+        log_mock.called
+
+        self.assertRaises(AssertionError, test_sweep.update_place_group, place,
+                          person_list=[person], person_weights=[],
+                          group_index=1)
+
+    @mock.patch("pyEpiabm.sweep.UpdatePlaceSweep.update_place_group")
+    def test__call__(self, mock_update):
         """Test whether the update place sweep function takes an
         initially empty place and correctly adds a person to
         the place.
         """
-        test_sweep = pe.sweep.UpdatePlaceSweep()
-        test_sweep.bind_population(self.pop)
-        self.assertFalse(self.place.persons)
-
-        self.place.add_person(self.person)
-        test_sweep(self.time)
-        self.assertTrue(self.place.persons)
-        self.place.add_person(self.person)
-        test_sweep(self.time)
-        self.assertEqual(len(self.place.persons), 1)
+        test_pop = self.pop
+        place = test_pop.cells[0].places[0]
+        person = test_pop.cells[0].persons[0]
+        mock_update.return_value = None
+        test_sweep = UpdatePlaceSweep()
+        test_sweep.bind_population(test_pop)
+        self.assertFalse(place.persons)
+
+        place.place_type = PlaceType.OutdoorSpace
+        test_sweep(1)
+        mock_update.called_with(place)
```

Comment:
I can't find anything about this syntax online - is this meant to me `assert_called_with()`